orderer/consensus/etcdraft/chain.go

```diff
@@ -1363,6 +1411,45 @@ func (c *Chain) newConfigMetadata(block *common.Block) *etcdraft.ConfigMetadata
 	return metadata
 }
 
+// validateTransitionToSmartbft validates the consensus type migration to smartbft, if any.
+// If the new config changes the consensus type to smartbft, it should have empty metadata.
+// Returns
+//
+//	-- transitionTobft bool; true if the new order config has a consensustype transition to smartbft,
+//	-- err error; the error in config metadata for smartbft, if any.
+//
+// It returns (true, nil) in the case of a valid config for transition to smartbft and (true, error) in case
+// the new config has a consensus type change to smartbft, but has invalid metadata.
+// It returns (false, nil) in case the new config doesn't change consensus type to smartbft.
+func (c *Chain) validateTransitionToSmartbft(oldOrdererConfig, newOrdererConfig channelconfig.Orderer) (bool, error) {
+	var transitionTobft bool
+	oldCapabilities := oldOrdererConfig.Capabilities()
+	if oldCapabilities == nil {
+		return transitionTobft, nil
+	}
+	if oldCapabilities.ConsensusTypeMigration() {
```

Comment:
you also need to check ConsensusTypeBFT() capability

Reply:
sure